### PR TITLE
Problem with Custom Field in Contact Reports

### PR DESCRIPTION
**Pull request by giovannidalmas1967**
Custom fields with data type 'String' cannot be displayed in Reports->Contacts->Contituent Summary->Summery Fields->Filters Tab. This should be related to function addCustomDataToColumns(). Putting a different control for variable '$option' in case 'String' seems to solve the issue

Overview
----------------------------------------
Problems with displaying Custom fields in Reports->Contacts. The code modified should solve this issue

Before
----------------------------------------
Custom fields with data type 'String' cannot be displayed in Reports->Contacts->Contituent Summary->Summery Fields->Filters Tab

After
----------------------------------------
Custom fields with data type 'String' now are visible in Reports->Contacts->Contituent Summary->Summery Fields->Filters Tab

Technical Details
----------------------------------------
Putting a different control for variable '$options' seems to solve the issue.

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@yashodha are you able to take a look at this?

**Comment by eileenmcnaughton** [comment]
add to whitelist

**Comment by eileenmcnaughton** [comment]
@giovannidalmas1967 if you click through on the 'details' above you'll see some style issues with your commit https://test.civicrm.org/job/CiviCRM-Core-PR/23002/checkstyleResult/new/ - we use drupal whitespace code standards - follow the links in https://github.com/civicrm/civicrm-core/pull/13044#issuecomment-434789334 for more

File: CRM/Report/Form.php [hunk 1]
```diff
@@ -3935,7 +3935,7 @@ public function addCustomDataToColumns($addFields = TRUE, $permCustomGroupIds =
           $curFilters[$fieldName]['type'] = CRM_Utils_Type::T_STRING;
 
           $options = CRM_Core_PseudoConstant::get('CRM_Core_BAO_CustomField', 'custom_' . $customDAO->cf_id, array(), 'search');
-          if ($options !== FALSE) {
+          if ((is_array($options) && count($options) != 0) || (!is_array($options) && $options !== FALSE)) {
```
Comment (giovannidalmas1967) [inline CRM/Report/Form.php]: @eileenmcnaughton  I added a couple of spaces, one before "!=! and the other after "!="

**Comment by alifrumin** [comment]
I was able to replicate this bug on a test site by:

1. creating a custom field on the individual that is searchable and is of the Data and Input Field Type Alphanumeric Text.
2. Going to CiviCRM Administer-> CiviReport -> Create Report from Template and creating a report using the template "Constituent Report (Summary)"
3. Going to the Filters tab of this report

On the filters tab for the  Constituent Report (Summary) there is no way to filter by the custom field that has Data and Input Field Type Alphanumeric Text.

Then I was able to confirm that this pr fixes that bug (with it one can filter by a custom field with the Data and Input Field Type Alphanumeric Text) by following the same steps on the Jenkins build.

This PR is ready to be merged from my perspective.

**Comment by eileenmcnaughton** [comment]
Thanks @alifrumin  - merging

**Comment by eileenmcnaughton** [comment]
Thanks for the patch @giovannidalmas1967

**Comment by giovannidalmas1967** [comment]
@eileenmcnaughton glad to be able to help :)

**Pull request merged**